Hashtable.c

```diff
@@ -79,10 +136,49 @@ void Hashtable_delete(Hashtable* this) {
    free(this);
 }
 
-void Hashtable_put(Hashtable* this, unsigned int key, void* value) {
+void Hashtable_setSize(Hashtable* this, unsigned int size) {
+
+   assert(Hashtable_isConsistent(this));
+
+   unsigned int newSize = nextPrime(size);
+   HashtableItem** newBuckets = (HashtableItem**) xCalloc(newSize, sizeof(HashtableItem*));
+
+   // rehash
+   for (unsigned int i = 0; i < this->size; i++) {
+      HashtableItem* walk = this->buckets[i];
+      while (walk) {
+         HashtableItem* next = walk->next;
+         walk->next = NULL;
+
+         unsigned int newIndex = walk->key % newSize;
+         HashtableItem** newBucketPtr = &newBuckets[newIndex];
+         while (*newBucketPtr)
+            newBucketPtr = &(*newBucketPtr)->next;
+         *newBucketPtr = walk;
+
+         walk = next;
+      }
+   }
+
+   this->size = newSize;
+   free(this->buckets);
+   this->buckets = newBuckets;
+
+   assert(Hashtable_isConsistent(this));
+}
+
+void Hashtable_put(Hashtable* this, hkey_t key, void* value) {
+
+   assert(Hashtable_isConsistent(this));
+   assert(this->size > 0);
+
+   // grow on load > 1.0
```

Comment:
`load` is a well defined term for a tool that is used in systems monitoring, better use `fill level` 1/2

Reply:
Load factor is a common term for describing hash tables too:
https://en.wikipedia.org/wiki/Hash_table#Dynamic_resizing